tests/cbmc/proofs/s2n_stuffer_reserve_space/s2n_stuffer_reserve_space_harness.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_mem.h"
+
+#include <sys/param.h>
+#include <assert.h>
+
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <cbmc_proof/make_common_datastructures.h>
+
+void s2n_stuffer_reserve_space_harness() {
+    /* Non-deterministic inputs. */
+    struct s2n_stuffer *stuffer = cbmc_allocate_s2n_stuffer();
+    __CPROVER_assume(s2n_stuffer_is_valid(stuffer));
+    uint32_t size;
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
+    if(nondet_bool()) {
+        s2n_mem_init();
+    }
+
+    /* Save previous state. */
+    struct s2n_stuffer old_stuffer = *stuffer;
+    struct store_byte_from_buffer old_byte;
+    save_byte_from_blob(&stuffer->blob, &old_byte);
+
+    /* Operation under verification. */
+    if (s2n_stuffer_reserve_space(stuffer, size) == S2N_SUCCESS) {
+        assert(s2n_stuffer_is_valid(stuffer));
+        if(s2n_stuffer_space_remaining(stuffer) < size) {
+            assert(stuffer->blob.size == MAX(size - s2n_stuffer_space_remaining(stuffer), 1024));
+            assert(stuffer->blob.allocated >= size);
+        }
+    } else {
+        assert(stuffer->blob.size == old_stuffer.blob.size);
```

Comment:
didn't we have a shortcut for asserting stuffer equivalence?

Reply:
We do, but here we assert that all fields are equivalent, but the `stuffer->blob.data`.

Reply:
Ah right. Maybe add a comment saying why we don't assert `blob.data` equivalence

Reply:
I added a comment as suggested, is it clear enough?